src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -128,24 +128,60 @@ public static string GetFullPath(string path, string basePath)
 
         public static string GetTempPath()
         {
-            StringBuilder sb = StringBuilderCache.Acquire(Interop.Kernel32.MAX_PATH);
-            uint r = Interop.Kernel32.GetTempPathW(Interop.Kernel32.MAX_PATH, sb);
-            if (r == 0)
+            Span<char> initialBuffer = stackalloc char[PathInternal.MaxShortPath];
+            ValueStringBuilder builder = new ValueStringBuilder(initialBuffer);
```

Comment:
Nit, you could use var.

Reply:
I believe the guideline is that we should use `var` whenever the type can be statically inferred from the right hand side. In most cases, that means, wherever the `new` keyword is used, we likely have a candidate for use of `var` (or when there is explicit casting). I would be curious to see in what instance you got feedback contrary to that.

Reply:
The guideline doesn't mandate it. https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md

Additionally I don't think the guideline is clear enough. It borders on gratuitous in some cases (`var values = new int[30]`, `var info = new FileInfo(path)`). This particular instance is a little grey, so I'm ok with var, but I don't think it is necessary. If I make another update I'll change it.